### PR TITLE
doc: release notes: Update contents for 2.2.0

### PR DESCRIPTION
**Pull request by jhedberg**
Add updated issue list, update conf.py with the new version, and
update the VERSION file

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

### CODE REVIEW

**Comment by jhedberg** [comment]
To be merged after #23357

**Review by carlescufi — COMMENTED** [review]
Why did you change permissions on the release notes rst file?

**Comment by jhedberg** [comment]
> Why did you change permissions on the release notes rst file?

That was unintentional. Probably the result of a shell redirect of the issue list into the file. Will fix.

**Comment by jhedberg** [comment]
>> Why did you change permissions on the release notes rst file?

> That was unintentional. Probably the result of a shell redirect of the issue list into the file. Will fix. 

Actually, I didn't change the permissions. Why do you say I did?

File: VERSION [hunk 1]
```diff
@@ -2,4 +2,4 @@ VERSION_MAJOR = 2
 VERSION_MINOR = 2
 PATCHLEVEL = 0
 VERSION_TWEAK = 0
-EXTRAVERSION = rc3
+EXTRAVERSION =
```
Comment (ioannisg) [inline VERSION]: I tihnk you need a separate PR for this change.

Comment (jhedberg) [reply VERSION]: I was just  following what what was done for 2.1 with commit b413223a66ed9e857c67ee21d1e8be4768804dd1

File: doc/conf.py [hunk 2]
```diff
@@ -527,6 +527,7 @@
     'theme_logo_only': False,
     'current_version': version,
     'versions': (("latest", "/"),
+                 ("2.2.0", "/2.2.0/"),
```
Comment (ioannisg) [inline doc/conf.py]: can you also remove one from the bottom here, so we got to display the same number of releases?

File: VERSION (hunk 1 above)
Comment (jhedberg) [reply VERSION]: Anyway, why would a separate PR be needed?

File: doc/conf.py (hunk 2 above)
Comment (jhedberg) [reply doc/conf.py]: sure

Comment (jhedberg) [reply doc/conf.py]: actually, I could remove two versions since 2.1 didn't remove any

Comment (jhedberg) [reply doc/conf.py]: or perhaps even three, so we start the list with the LTS release?

File: VERSION (hunk 1 above)
Comment (ioannisg) [reply VERSION]: Well, actually, a separate commit (but then a separate PR would also make CI run once more).
You should update the version file with a commit with "release: " in the title, and appropriate content.
https://github.com/zephyrproject-rtos/zephyr/wiki/Release-procedure

**Comment by jhedberg** [comment]
@ioannisg I've now split the PR into three commits, one for each file and with the last commit updating `VERSION`. I also removed all 1.x versions from `doc/conf.py` except for the last one, i.e. the LTS release.

File: doc/conf.py (hunk 2 above)
Comment (ioannisg) [reply doc/conf.py]: that's also fine with me.

**Review by ioannisg — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**